pkg/apis/policy/v1beta1/clusterimagepolicy_validation.go

```diff
@@ -219,6 +224,20 @@ func (source *Source) Validate(ctx context.Context) *apis.FieldError {
 	var errs *apis.FieldError
 	if source.OCI == "" {
 		errs = errs.Also(apis.ErrMissingField("oci"))
+	} else {
+		// We want to validate both registry uris only or registry with valid repository names
```

Comment:
I'm fine with this small change but I think I'd still prefer to pull this into a function since it's still cut&paste in both places :) But I'm not going to block it. I'm just thinking that once we roll to a new version later on, we'll have to do this then too, so might as well do it now. But up to you :)

Reply:
I'll take care of that in a different PR so I can merge these changes now.

Reply:
I've addressed here: https://github.com/sigstore/policy-controller/pull/341